include/picongpu/particles/collision/binary/DefaultAlg.hpp

```diff
@@ -0,0 +1,494 @@
+#include <utility>
+#include <type_traits>
+#include <cmath>
+#include <stdio.h>
+#include <fenv.h>
+
+namespace picongpu
+{
+namespace particles
+{
+namespace collision
+{
+namespace binary
+{
+namespace acc
+{
+    using namespace pmacc;
+    struct DefaultAlg
+    {
+        float_X densitySqCbrt0;
+        float_X densitySqCbrt1;
+        uint32_t duplications;
+        uint32_t potentialPartners;
+        float_X coulombLog;
+
+        //! store user manipulators instance
+        HDINLINE DefaultAlg(
+            float_X p_densitySqCbrt0,
+            float_X p_densitySqCbrt1,
+            uint32_t p_potentialPartners,
+            float_X p_coulombLog
+        ):
+        densitySqCbrt0( p_densitySqCbrt0 ),
+        densitySqCbrt1( p_densitySqCbrt1 ),
+        duplications( 1u ),
+        potentialPartners( p_potentialPartners ),
+        coulombLog( p_coulombLog )
+        { };
+
+        static constexpr float_X c = SPEED_OF_LIGHT;
+
+        static DINLINE float3_X labToComs(
+            float3_X labMomentum,
+            float_X mass,
+            float_X gamma,
+            float_X gammaComs,
+            float_X factorA,
+            float3_X comsVelocity
+        )
+        {
+            float_X dot = pmacc::math::dot(comsVelocity, labMomentum);
+            float_X factorB = mass * gamma * gammaComs;
+            float3_X diff = ( factorA * dot - factorB ) * comsVelocity;
+            return labMomentum + diff;
+        }
+
+        // (13) in [Perez 2012]
+        static DINLINE float3_X comsToLab(
+            float3_X comsMomentum,
+            float_X mass,
+            float_X coeff,
+            float_X gammaComs,
+            float_X factorA,
+            float3_X comsVelocity
+        )
+        {
+            float_X dot = pmacc::math::dot(comsVelocity, comsMomentum);
+            float_X factorB = coeff * gammaComs;
+            float3_X diff = ( factorA * dot + factorB ) * comsVelocity;
+            return comsMomentum + diff;
+        }
+
+        static DINLINE float_X calcRelativeComsVelocity(
+            float_X comsMomentumMag0,
+            float_X mass0,
+            float_X mass1,
+            float_X gamma0,
+            float_X gamma1,
+            float_X coeff0,
+            float_X coeff1,
+            float_X gammaComs
+        )
+        {
+            float_X val = ( mass0 * gamma0 + mass1 * gamma1 )
+                * comsMomentumMag0;
+            val = val / ( coeff0 * coeff1 * gammaComs );
+            return val;
+
+        }
+        // Stared gamma times mass.
+        static DINLINE float_X coeff(
+            float3_X labMomentum,
+            float_X mass,
+            float_X gamma,
+            float_X gammaComs,
+            float3_X comsVelocity
+        )
+        {
+            float_X dot = pmacc::math::dot( comsVelocity, labMomentum );
+            float_X val = mass * gamma - dot / ( c*c );
+            return gammaComs * val;
+        }
+
+        static DINLINE float_X calcCosXi(
```

Comment:
please add a docstring what this function is doing. If the `s12` and `u` are limited to a defined range please add the valid range too.